diesel/src/migrations/migration.rs

```diff
@@ -0,0 +1,166 @@
+use connection::Connection;
+use super::{MigrationError, RunMigrationsError};
+
+use std::path::{Path, PathBuf};
+
+pub trait Migration {
+    fn version(&self) -> i64;
```

Comment:
I have one concern with that. If we assume it's always a number, we have a clearly defined way to determine what is and isn't a migration in this directory (for example, we shouldn't error when we see `.gitkeep`). We'd need to make sure we have a strict definition, as I do want to error in the case where something looks an awful lot like a migration, but is off somehow (`down.sql` being missing for example). Is ignoring files with a leading `.` enough here?